scripts/cat/save_load.py

```diff
@@ -0,0 +1,151 @@
+"""List of IDs of faded cats"""
+
+cat_to_fade = []
+"""Cats who have been faded since the last save"""
+
+
+def save_cats(clanname, cat_class: Type["Cat"], game: "Game"):
+    """Save the cat data."""
+
+    directory = get_save_dir() + "/" + clanname
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+    # Delete all existing relationship files
+    if not os.path.exists(directory + "/relationships"):
+        os.makedirs(directory + "/relationships")
+    for f in os.listdir(directory + "/relationships"):
+        os.remove(os.path.join(directory + "/relationships", f))
+
+    save_faded_cats(clanname, cat_class, game)  # Fades cat and saves them, if needed
+
+    clan_cats = []
+    for inter_cat in cat_class.all_cats.values():
+        cat_data = inter_cat.get_save_dict()
+        clan_cats.append(cat_data)
+
+        inter_cat.save_condition()
+
+        if inter_cat.history:
+            inter_cat.save_history(directory + "/history")
+            # after saving, dump the history info
+            inter_cat.history = None
+        if not inter_cat.dead:
+            inter_cat.save_relationship_of_cat(directory + "/relationships")
+
+    safe_save(f"{get_save_dir()}/{clanname}/clan_cats.json", clan_cats)
+
+
+def save_faded_cats(clanname, cat_class: Type["Cat"], game: "Game"):
+    """Deals with fades cats, if needed, adding them as faded"""
+    global cat_to_fade
+
+    if cat_to_fade:
+        directory = get_save_dir() + "/" + clanname + "/faded_cats"
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+    copy_of_info = ""
+    for cat in cat_to_fade:
+        inter_cat = cat_class.all_cats[cat]
+
+        # Add ID to list of faded cats.
+        faded_ids.append(cat)
+
+        # If they have a mate, break it up
+        if inter_cat.mate:
+            for mate_id in inter_cat.mate:
+                if mate_id in cat_class.all_cats:
+                    cat_class.all_cats[mate_id].unset_mate(inter_cat)
+
+        # If they have parents, add them to their parents "faded offspring" list:
+        for x in inter_cat.get_parents():
+            if x in cat_class.all_cats:
+                cat_class.all_cats[x].faded_offspring.append(cat)
+            else:
+                parent_faded = add_faded_offspring_to_faded_cat(clanname, x, cat)
+                if not parent_faded:
+                    print(f"WARNING: Can't find parent {x} of {cat.name}")
+
+        # Get a copy of info
+        if get_game_setting("save_faded_copy"):
+            copy_of_info += (
+                ujson.dumps(inter_cat.get_save_dict(), indent=4)
+                + "\n--------------------------------------------------------------------------\n"
+            )
+
+        # SAVE TO ITS OWN LITTLE FILE. This is a trimmed-down version for relation keeping only.
+        cat_data = inter_cat.get_save_dict(faded=True)
+
+        safe_save(f"{get_save_dir()}/{clanname}/faded_cats/{cat}.json", cat_data)
```

Comment:
Moving forward, we should try to use Pathlib instead of constructing path literals as strings.